plugins/module_utils/network/vyos/facts/bgp_global/bgp_global.py

```diff
@@ -59,7 +59,9 @@ def populate_facts(self, connection, ansible_facts, data=None):
             if "address-family" not in resource:
                 config_lines.append(re.sub("'", "", resource))
 
-        bgp_global_parser = Bgp_globalTemplate(lines=config_lines)
+        bgp_global_parser = Bgp_globalTemplate(
```

Comment:
The `__init__()` in templates needs to be updated to accept module object.
For example: https://github.com/ansible-collections/cisco.nxos/blob/main/plugins/module_utils/network/nxos/rm_templates/bgp_global.py#L58